packages/backend-core/src/objectStore/objectStore.ts

```diff
@@ -146,12 +173,20 @@ export async function upload({
   path,
   type,
   metadata,
+  body,
+  addTTL,
 }: UploadParams) {
   const extension = filename.split(".").pop()
-  const fileBytes = fs.readFileSync(path)
+
+  const fileBytes = path ? (await fsp.open(path)).createReadStream() : body
 
   const objectStore = ObjectStore(bucketName)
-  await makeSureBucketExists(objectStore, bucketName)
+  const bucketCreated = await createBucketIfNotExists(objectStore, bucketName)
+
+  if (addTTL && bucketCreated.created) {
+    let ttlConfig = bucketTTLConfig(bucketName, 1)
```

Comment:
If I get this right, we will never be able to set the TTL is the bucket already exists